app/questionnaire/router.py

```diff
@@ -142,7 +132,16 @@ def get_first_incomplete_location_in_survey_url(self):
             )
             return self.get_section_resume_url(section_routing_path)
 
-        return self.get_last_location_in_survey().url()
+        return self.get_next_location_url_for_questionnaire_flow()
+
+    def get_next_location_url_for_questionnaire_flow(self) -> str:
+        if self._schema.is_questionnaire_flow_hub and self.can_access_hub():
+            return url_for("questionnaire.get_questionnaire")
```

Comment:
Correct, but it is needed on the endpoint as a user can explicitly visit it. It is needed here to prevent an unnecessary redirect which would go to the hub then redirect to the first incomplete location.